run.py

```diff
@@ -33,17 +37,80 @@ def _get_opt_parser(self):
         parser.add_option('-p', '--pass', dest='password', default=None)
         return parser
 
-    def _get_connection(self):
+    def _get_connection(self, hostname, username, password):
```

Comment:
Yeah I will include the --provider flag as you mentioned in the cards, will need to see how juju ssh unit works.